lib/fhir/core.js

```diff
@@ -236,29 +236,50 @@ module.exports = (mongo, modules) => {
                       return callback(err)
                     }
 
-                    const response = fhirCommon.bundleResults('searchset', results, total, ctx.url)
-                    if (_getpagesoffset + _count < total) {
-                      fhirCommon.addBundleLinkNext(response, ctx.url, _getpagesoffset + _count, _count)
-                    }
-
-                    hooks.executeAfterHooks('search', fhirModule, ctx, resourceType, response, (err, badRequest, results) => {
-                      if (err || badRequest) {
-                        return handleErrorAndBadRequest(err, badRequest, callback)
-                      }
-
-                      // format the updated entry array for the response
-                      if (results && results.entry) {
-                        results.entry.forEach((resource) => {
-                          fhirCommon.formatResource(resource.resource)
-                        })
-                      }
-
-                      if (ctx.operation === '$ihe-pix') {
-                        return callback(null, { httpStatus: ctx.responseStatus, resource: results })
-                      }
-
-                      callback(null, { httpStatus: 200, resource: results })
-                    })
+                    fhirCommon.mapSearchNameToPath(ctx.query._include)
+                      .then((paths) => {
+                        fhirCommon.includeResources(paths, results)
```

Comment:
This promise should be returned so that it can be chained and so that we don't need multiple catch blocks.